pkg/segment/sortindex/sortindex.go

```diff
@@ -658,37 +657,85 @@ func readMetadata(file *os.File) (*metadata, error) {
 	return meta, nil
 }
 
-func SetSortColumns(columnNames []string) error {
-	sortConfig.mu.Lock()
-	defer sortConfig.mu.Unlock()
+func SetSortColumns(indexName string, columnNames []string) error {
+	sortConfigMutex.Lock()
+	defer sortConfigMutex.Unlock()
+
+	dir := filepath.Dir(SortColumnsConfigPath)
+	if err := os.MkdirAll(dir, 0755); err != nil {
+		return fmt.Errorf("failed to create config directory: %v", err)
+	}
 
-	// TODO: Persist sort column names to disk
 	for _, col := range columnNames {
 		if col == "" {
 			return fmt.Errorf("SetSortColumns: column names must be non-empty strings")
 		}
 	}
 
-	sortConfig.columns = columnNames
+	config := SortColumnsConfig{
+		Indexes: make(map[string][]string),
+	}
+
+	data, err := os.ReadFile(SortColumnsConfigPath)
+	if err == nil {
+		if err := json.Unmarshal(data, &config); err != nil {
+			return fmt.Errorf("failed to parse sort columns config: %v", err)
+		}
+	}
+
+	config.Indexes[indexName] = columnNames
+
+	data, err = json.MarshalIndent(config, "", "    ")
+	if err != nil {
+		return fmt.Errorf("failed to marshal sort columns config: %v", err)
+	}
+
+	err = os.WriteFile(SortColumnsConfigPath, data, 0644)
+	if err != nil {
+		return fmt.Errorf("failed to write sort columns config: %v", err)
+	}
 
 	return nil
 }
 
-func GetSortColumns() []string {
-	sortConfig.mu.RLock()
-	defer sortConfig.mu.RUnlock()
-	return sortConfig.columns
+func GetSortColumnNamesForIndex(indexName string) []string {
+	sortConfigMutex.RLock()
+	defer sortConfigMutex.RUnlock()
+
+	data, err := os.ReadFile(SortColumnsConfigPath)
+	if err != nil {
+		log.Debugf("No sort columns config file found: %v", err)
+		return make([]string, 0)
+	}
```

Comment:
If the file doesn't exist, this handling looks good. But if there's any other error (e.g., permissions), then you should log that as an error